src/Timer/Timers.php

```diff
@@ -18,10 +18,17 @@ final class Timers
     private $timers = array();
     private $schedule = array();
     private $sorted = true;
+    private $useHighResolution;
+
+    public function __construct()
+    {
+        // prefer high-resolution timer, available as of PHP 7.3+
+        $this->useHighResolution = \function_exists('hrtime');
+    }
 
     public function updateTime()
     {
-        return $this->time = \microtime(true);
+        return $this->time = $this->useHighResolution ? \hrtime(true) * 1e-9 : \microtime(true);
```

Comment:
> `* 1e-9`

What is the reasoning for doing that?

Reply:
On 64bit `hrtime` returns an integer - on 32bit a float. So the idea is to get a float just like `microtime`, but on 32bit this will break the timer.

Reply:
Is there a way we can test this on both 64 and 32 bit on Travis? Just to be sure we don't break anything for our users

Reply:
This converts from nanoseconds (which could be given an `int` or `float` depending on architecture) to a `float` second value (`1e-9` means `10^-9` because 1s = 10^9ns). Even on 32 bit platforms, a `float` value in PHP usually follows the 64 bit IEEE format. (http://php.net/manual/en/language.types.float.php and https://stackoverflow.com/questions/1101463/the-double-byte-size-in-32-bit-and-64-bit-os).

In other words: We have no reason to believe that this wouldn't work on any system that this project currently works on. We already use `float` values to store the timestamps in other places in this very project.

Reply:
Thank you for that explanation @clue :+1: